src/DependencyInjection/Compiler/TemplatingValidatorPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2017 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\RoutingBundle\DependencyInjection\Compiler;
+
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Cmf\Bundle\RoutingBundle\Validator\Constraints\RouteDefaultsTemplatingValidator;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * To avoid a BC-Break: If templating component exists, we will use the validator using general templating
+ * from FrameworkBundle.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@mayflower.de>
+ */
+class TemplatingValidatorPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (interface_exists(EngineInterface::class) && $container->has('templating')) {
+            $templatingDefinition = $container->findDefinition('templating');
+            $validatorDefinition = $container->getDefinition('cmf_routing.validator.route_defaults');
+            $validatorDefinition->setClass(RouteDefaultsTemplatingValidator::class);
+            $validatorDefinition->replaceArgument(1, $templatingDefinition);
```

Comment:
this is wrong. You must not copy the definition in the argument. This would create a separate instance of the service. You must use `new Reference('templating')` as value of the argument, to reference the existing service

Reply:
@stof  but i would then inject it by `replaceArgument()` right? but thanks, had some trouble how to use aliases with service definitions.